cmake/sysbuild/suit.cmake

```diff
@@ -0,0 +1,377 @@
+  )
+  list(APPEND TEMPLATE_ARGS ${core_arguments})
+  list(APPEND TEMPLATE_ARGS --artifacts-folder "${SUIT_ROOT_DIRECTORY}")
+   set_property(
+    GLOBAL APPEND PROPERTY SUIT_POST_BUILD_COMMANDS
+    COMMAND ${PYTHON_EXECUTABLE} ${SUIT_GENERATOR_BUILD_SCRIPT}
+    template
+    ${TEMPLATE_ARGS}
+  )
+endfunction()
+
+# Create binary envelope from input yaml file.
+#
+# Usage:
+#   suit_create_envelope(<input_file> <output_file>)
+#
+# Parameters:
+#   'input_file' - path to input yaml configuration
+#   'output_file' - path to output binary suit envelope
+function(suit_create_envelope input_file output_file)
+  set_property(
+    GLOBAL APPEND PROPERTY SUIT_POST_BUILD_COMMANDS
+    COMMAND ${PYTHON_EXECUTABLE} ${SUIT_GENERATOR_CLI_SCRIPT}
+    create
+    --input-file ${input_file}
+    --output-file ${output_file}
+  )
+  sysbuild_get(CONFIG_SUIT_ENVELOPE_SIGN IMAGE ${DEFAULT_IMAGE} VAR CONFIG_SUIT_ENVELOPE_SIGN KCONFIG)
+  if (CONFIG_SUIT_ENVELOPE_SIGN)
+    suit_sign_envelope(${output_file} ${output_file})
+  endif()
+endfunction()
+
+# Copy artifact to the common SUIT output directory.
+#
+# Usage:
+#   suit_copy_artifact_to_output_directory(<target> <artifact>)
+#
+# Parameters:
+#   'target' - target name
+#   'artifact' - path to artifact
+function(suit_copy_artifact_to_output_directory target artifact)
+  cmake_path(GET artifact FILENAME artifact_filename)
+  cmake_path(GET artifact EXTENSION artifact_extension)
+  set(destination_name "${target}${artifact_extension}")
+  set_property(
+    GLOBAL APPEND PROPERTY SUIT_POST_BUILD_COMMANDS
+    COMMAND ${CMAKE_COMMAND} -E copy ${artifact} ${SUIT_ROOT_DIRECTORY}${target}.bin
+  )
+endfunction()
+
+# Register SUIT post build commands.
+#
+# Usage:
+#   suit_register_post_build_commands()
+#
+function(suit_register_post_build_commands)
+  get_property(
+    post_build_commands
+    GLOBAL PROPERTY
+    SUIT_POST_BUILD_COMMANDS
+  )
+
+  foreach(image ${IMAGES})
+    sysbuild_get(BINARY_DIR IMAGE ${image} VAR APPLICATION_BINARY_DIR CACHE)
+    sysbuild_get(BINARY_FILE IMAGE ${image} VAR CONFIG_KERNEL_BIN_NAME KCONFIG)
+    list(APPEND dependencies "${BINARY_DIR}/zephyr/${BINARY_FILE}.bin")
+  endforeach()
+
+  add_custom_target(
+    create_suit_artifacts
+    ALL
+    ${post_build_commands}
+    DEPENDS
+    ${dependencies}
+    COMMAND_EXPAND_LISTS
+    COMMENT "Create SUIT artifacts"
+  )
+endfunction()
+
+# Create DFU package/main envelope.
+#
+# Usage:
+#   suit_create_package()
+#
+function(suit_create_package)
+
+  #if(NOT EXISTS "${SUIT_ROOT_DIRECTORY}")
+    add_custom_target(
+      suit_prepare_output_folder
+      ALL
+      COMMAND ${CMAKE_COMMAND} -E make_directory ${SUIT_ROOT_DIRECTORY}
+      COMMENT
+      "Create DFU output directory"
+    )
+  #endif()
+  set(SUIT_OUTPUT_ARTIFACTS_ITEMS)
+  set(SUIT_OUTPUT_ARTIFACTS_TARGETS)
+  set(CORE_ARGS)
+  set(STORAGE_BOOT_ARGS)
+  if (!CONFIG_HW_REVISION_SOC1)
+    set(STORAGE_UPDATE_ARGS)
+  endif()
```

Comment:
why is this here?

Reply:
leftover from code copied from -next repository, thx for spotting that